### PR TITLE
Ao live test success

### PR DESCRIPTION
**Pull request by prhuft**
The AnalogOutput device succeeds in performing equivalently to when we are running the LabVIEW server. Specifically, after connecting CsPy to the server and updating the settings with cycle_continuously checked, the waveform is successfully written to hardware and output as expected. Toggling cycle_continuously in successive update settings events has the desired effect. 

Please checkout this branch and run the server to confirm nothing anomalous occurs on your experiment.

### CODE REVIEW

**Comment by JuanBohorquez3** [comment]
Entering "q" to quit the server locks it (only way to close it or do anything is to close the terminal)  in the following cases:
- Keylistener has been started but connection has not been established
- Connection to server has been opened, then closed.
Looks like the server doesn't like to quit when there is no established connection.
Last message printed to terminal (logger.info etc) is :

> The following devices will be stopped: [<analogout.AnalogOutput object at 0x10FB0220>]

I think this issue might be broader than just the AO code as I see it in the HSDIO.

File: analogout.py [hunk 1]
```diff
@@ -159,17 +159,12 @@ def init(self):
...
                     try:
-                        self.task.close()
-
-                    except DaqError:
-                        # end the task nicely
-                        self.stop()
                         self.close()
-                        msg = '\n AnalogOutput failed to close current task'
-                        raise HardwareError(self, task=self.task, message=msg)
+                    except DaqError: # could extract error code to check if stale reference
```
Comment (JuanBohorquez3) [inline analogout.py]: I would like this to test the code for stale reference. Or at least print the error so we don't have another failure mode go silent.

**Review by JuanBohorquez3 — CHANGES_REQUESTED** [review]
Works as expected but server breaks in corner case. I think the breaking is broader issue, from outside this PR.

Only change I want is on the error handling when closing the task.

**Comment by prhuft** [comment]
@JuanBohorquez3 The above issue of the server not quitting when a connection has not been established is a known issue. It is due to the fact that the socket.accept call which listens for a connection attempt is blocking. This can be fixed in another PR, in a hacky way by opening a local socket when 'q' is pressed (the "connects to itself") and then the shutdown will proceed. In the meanwhile, to avoid having to close the terminal, you can just open a connection from CsPy when the shutdown stalls.

**Comment by prhuft** [comment]
Reminder @c-poole please checkout and test this branch at soonest convenience.

**Review by JuanBohorquez3 — APPROVED** [review]
Looks good, works great on my machine!

**Comment by c-poole** [comment]
I will be able to test this tonight. The weather situation on Thursday made me flee from Chamberlin during the relatively calm period in between the heavy downcasts so I didn't have the time I hoped for to test things. I will get to both of the PRs tonight.

**Comment by c-poole** [comment]
It seems as though I accidentally tested the wrong branch with a similar name when I looked at this one. Now that I have resolved this, I am running into an issue launching the server. I installed the requirements.txt file with pip in a virtual environment and have version 0.5.7 and I'm running Python 3.8.2. Is this consistent with your machine?
image

**Comment by prhuft** [comment]
Cody's issue above was due to an error in my import statement. It should be fixed now.

**Comment by c-poole** [comment]
LGTM

**Review by c-poole — APPROVED** [review]

**Pull request merged**